smolyay/grid.py

```diff
@@ -0,0 +1,362 @@
+
+import abc
+import numpy
+from smolyay.basis import (BasisFunction, ChebyshevFirstKind,
+                           BasisFunctionSet, NestedBasisFunctionSet)
+import itertools
```

Comment:
Per PEP8, import order should be core libraries, third party libraries, then your application, with a line break between these groups.
```suggestion
import abc
import itertools

import numpy

from smolyay.basis import (BasisFunction, ChebyshevFirstKind,
                           BasisFunctionSet, NestedBasisFunctionSet)
```